models/dag_node.go

```diff
@@ -0,0 +1,47 @@
+package models
+
+import (
+	"time"
+
+	pb "github.com/odpf/optimus/api/proto/odpf/optimus"
+	"github.com/odpf/optimus/core/set"
+)
+
+// DAGNode represents a custom data type that contains a DAGSpec along with it's dependent DAGNodes
+type DAGNode struct {
+	DAG        JobSpec
```

Comment:
What if we make it generic and use an interface here? Just a suggestion, not sure. This can be moved to core then.

Reply:
This is a good idea, I have made this change